oshmem/mca/atomic/ucx/atomic_ucx_cswap.c

```diff
@@ -27,44 +27,40 @@ int mca_atomic_ucx_cswap(void *target,
                          int pe)
 {
     ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
     spml_ucx_mkey_t *ucx_mkey;
     uint64_t rva;
+    uint64_t val;
+    uint64_t cmp;
+
+    if (8 == nlong) {
+        val = *(uint64_t*)value;
```

Comment:
still 3 branches.. memcpy is also a branch
ok how about this:
```
static inline mca_atomic_ucx_cswap_do(..., int nlong, ...) {
 ... 
}

mca_atomic_ucx_cswap(...) {
 if (8 == nlong) {
   mca_atomic_ucx_cswap_do(..., 8, ...);
 } else if (4 == nlong) {
  mca_atomic_ucx_cswap_do(..., 4, ...);
 } else {
   error
 }
}
```

Reply:
@yosefe done